go/db/tls.go

```diff
@@ -35,12 +35,9 @@ import (
 const Error3159 = "Error 3159:"
 const Error1045 = "Access denied for user"
 
-// Track if a TLS has already been configured for topology
-var topologyTLSConfigured bool = false
-
-// Track if a TLS has already been configured for Orchestrator
-var orchestratorTLSConfigured bool = false
-
+// This cache exists to prevent re-registering tls.Conf instances while also
+// allowing them to be interrogated in tests.
+var tlsConfigCache *cache.Cache = cache.New(cache.NoExpiration, time.Second)
```

Comment:
Being not _terribly_ familiar with Golang, this seemed to me as the best option to be able to inspect the `tls.Config` objects generated in the methods. I figure that the ability to test the code outweighs the downside of a couple structs lingering in memory. I'm happy to change this if there's a better way.